src/components/periodic-report/periodic-report.repository.ts

```diff
@@ -22,50 +24,36 @@ import {
   PeriodicReportListInput,
   ProgressReport,
   ReportType,
+  resolveReportType,
 } from './dto';
 
 @Injectable()
 export class PeriodicReportRepository extends DtoRepository(IPeriodicReport) {
   async create(input: CreatePeriodicReport) {
-    const id = await generateId();
-    const createdAt = DateTime.local();
     const reportFileId = await generateId();
 
+    const Report = resolveReportType(input);
+    const initialProps = {
+      type: input.type,
+      start: input.start,
+      end: input.end,
+      receivedDate: null,
+      reportFile: reportFileId,
+    };
     const query = this.db
       .query()
-      .create([
-        [
-          node(
-            'newPeriodicReport',
-            ['PeriodicReport', 'BaseNode', `${input.type}Report`],
-            {
-              createdAt,
-              id,
-            }
-          ),
-        ],
-        ...property('type', input.type, 'newPeriodicReport'),
-        ...property('start', input.start, 'newPeriodicReport'),
-        ...property('end', input.end, 'newPeriodicReport'),
-        ...property('receivedDate', null, 'newPeriodicReport'),
-        ...property('reportFile', reportFileId, 'newPeriodicReport'),
-      ])
-      .with('newPeriodicReport')
-      .match(node('parent', 'BaseNode', { id: input.projectOrEngagementId }))
-      .create([
-        node('parent'),
-        relation('out', '', 'report', {
-          active: true,
-          createdAt,
-        }),
-        node('newPeriodicReport'),
-      ])
-      .return('newPeriodicReport.id as id');
+      .apply(await createNode(Report, { initialProps }))
```

Comment:
Does createNode add `PeriodicReport` and the specific report type label (i.e. `ProgressReport`)?

Reply:
`BaseNode labels are pulled from the resource's class hierarchy or manually`. That sounds like it might do the trick.

Reply:
Yes and yes